docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+                from_treasury = int32;
+                to_principal = opt principal;
+                to_subaccount = opt record { subaccount = vec nat8 };
+                memo = opt nat64;
+                amount_e8s = nat64;
+               
+            };
+        };
+    };
+)'
+```
+
+## `UpgradeSnsControlledCanister` 
+
+### Relevant type signatures
+
+```candid
+    type UpgradeSnsControlledCanister : UpgradeSnsControlledCanister;
+
+    type UpgradeSnsControlledCanister = record {
+        new_canister_wasm : vec nat8;
+        mode : opt int32;
+        canister_id : opt principal;
+        canister_upgrade_arg : opt vec nat8;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            UpgradeSnsControlledCanister = record {
+                new_canister_wasm = vec nat8;
+                mode = opt int32;
+                canister_id = opt principal;
+                canister_upgrade_arg = opt vec nat8;
+            };
+        };
+    };
+)'
+```
+
+## `DeregisterDappCanisters` 
+
+### Relevant type signatures
+
+```candid
+    type DeregisterDappCanisters : DeregisterDappCanisters;
+
+    type DeregisterDappCanisters = record {
+        canister_ids : vec principal;
+        new_controllers : vec principal;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            DeregisterDappCanisters = record {
+                canister_ids = vec principal;
+                new_controllers = vec principal;
+            };
+        };
+    };
+)'
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            <PROPOSAL_TYPE> = <PARAMETERS_OF_PROPOSAL_TYPE>
+        };
+    }
+)'
+```
+
+## `ManageSnsMetadata` 
+
+### Relevant type signatures
+
+```candid
+    type  ManageSnsMetadata : ManageSnsMetadata;
+
+    type ManageSnsMetadata = record {
+        url : opt text;
+        logo : opt text;
+        name : opt text;
+        description : opt text;
+    };
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            <PROPOSAL_TYPE> = <PARAMETERS_OF_PROPOSAL_TYPE>
+        };
+    }
+)'
+```
+
+## `ExecuteGenericNervousSystemFunction` 
+
```

Comment:
```suggestion
After a generic proposal has been registered with a  `AddGenericNervousSystemFunction` proposal, such a proposal can be submitted with a `ExecuteGenericNervousSystemFunction` proposal.
The proposal identifies the previously added generic proposal by an ID (so called `function_id`) and, in addition, defines a payload.
Upon submission of such a proposal, the defined validation method is called, which checks that the given payload is valid for this kind of proposal (the method that will be called for this is the method `validator_method_name` on the canister `validator_canister_id` as it was defined when the generic proposal was added. If this validation is successful, the proposal will be created.

Later, if the proposal is adopted, the SNS governance canister will call the method `target_method_name ` on the canister `target_canister_id` (as also define in the generic proposal) with the payload defined here.
```

Due to this I think it would be helpful to have the proposals for adding, removing, and executing generic proposals together. It will then make sense what each of the fields mean. Maybe there could even be one part "relevant type signatures" that summarizes it all.